### PR TITLE
Improves LTT AP files messaging

### PR DESCRIPTION
**Pull request by evanchia-ly-sdets**
Signed-off-by: evanchia-ly-sdets <evanchia@amazon.com>

## What does this PR do?
Improves exception messaging and code clean up in LyTestTools AssetProcessor files. Removes redundant or unused functions as well.

## How was this PR tested?
Ran smoke and main test suites locally

### CODE REVIEW

File: Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def wait_for_idle(self, timeout=DEFAULT_TIMEOUT_SECONDS):
 
         self.send_message("waitforidle")
         result = self.read_message(read_timeout=timeout)
-        assert self.process_exists(), "Asset Processor appears unexpectedly shut down, or has crashed"
+        assert self.process_exists(), "Asset Processor has unexpectedly shut down during idle wait!"
```
Comment (Kadino) [inline Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py:114]: I don't see how this message is clearer. Removing suggestion of a possible crash seems to make it less helpful. Also the exception will already include wait_for_idle in its stacktrace.

minor: exclamation marks in error messages can feel a bit aggressive / annoying when debugging.

File: Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py [hunk 2]
```diff
@@ -129,7 +129,7 @@ def next_idle(self):
 
         self.send_message("signalidle")
         result = self.read_message()
-        assert self.process_exists(), "Asset Processor appears unexpectedly shut down, or has crashed"
+        assert self.process_exists(), "Asset Processor has unexpectedly shut down during idle request!"
```
Comment (Kadino) [inline Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py:132]: ditto

File: Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py [hunk 3]
```diff
@@ -209,7 +209,7 @@ def _get_port_from_log():
                     logger.debug(f"Read port type {port_type} : {port}")
                     return True
                 except Exception as ex:  # intentionally broad
-                    logger.debug("Failed to read port from file", exc_info=ex)
+                    logger.warning(f"Failed to read port type {port_type} : {port} from file", exc_info=ex)
```
Comment (Kadino) [inline Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py:212]: I seem to remember this message was set to debug to reduce spam. Line 220 can result in many read attempts before the expected line get written to this file.

File: Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py [hunk 4]
```diff
@@ -274,7 +274,8 @@ def _attempt_connection():
                             logger.debug(f"Found new connect port for {port_name}: {host}:{new_connect_port}")
                             connect_port = new_connect_port
                     except Exception as read_exception:  # Purposefully broad
-                        logger.debug(f"Failed to read port data", exc_info=read_exception)
+                        logger.warning(f"Failed to read port data for {port_name}: {host}:{new_connect_port}",
+                                       exc_info=read_exception)
```
Comment (Kadino) [inline Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py:278]: ditto for line 282

File: Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py [hunk 5]
```diff
@@ -654,7 +655,9 @@ def set_failure_log_folder(self, log_root):
     def check_copy_logs(self):
         if self._temp_asset_root and self._failed_log_root:
             source_path = os.path.join(self._temp_asset_root, "logs")
-            dest_path = os.path.join(self._failed_log_root, f"{self._function_name + '.' if self._function_name else ''}{int(round(time.time() * 1000))}")
+            dest_path = os.path.join(self._failed_log_root,
+                                     f"{self._function_name + '.' if self._function_name else ''}"
+                                     f"{int(round(time.time() * 1000))}")
```
Comment (Kadino) [inline Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py:660]: The line break between 659 and 660 makes this less readable to me, prompting me think about how the precedence will work between if/else and string concatenation.

Recommend resolving the if/else outside of the call to join and storing it in a variable with a name similar to `failed_log_folder`

File: Tools/LyTestTools/ly_test_tools/o3de/asset_processor_config_util.py [hunk 6]
```diff
@@ -66,28 +65,13 @@ def enable_platform(config_setreg_path, platform):
...
     """
-    logger.debug("Enabling platform '{0}' in '{1}'".format(platform, AssetProcessorConfig))
+    logger.debug(f"Enabling platform '{platform}' in '{AssetProcessorConfig}'")
 
     file_location = path.join(config_setreg_path, AssetProcessorConfig)
     with RegistrySettings(file_location) as settings:
         settings.set_key(f'{ASSET_PROCESSOR_SETTINGS_ROOT_KEY}/Platforms/{platform}', 'enabled')
 
 
-def enable_all_platforms(config_setreg_path):
```
Comment (Kadino) [inline Tools/LyTestTools/ly_test_tools/o3de/asset_processor_config_util.py:76]: These utilities tend to be used in other modules. Is this safe to remove? Are there no calls to this function anywhere in the O3DE codebase?

File: Tools/LyTestTools/ly_test_tools/o3de/asset_processor_config_util.py [hunk 7]
```diff
@@ -98,30 +82,14 @@ def disable_platform(config_setreg_path, platform):
...
+    logger.debug(f"Disabling platform '{platform}' in '{AssetProcessorConfig}'")
 
-    file_location = path.join(config_setreg_path, AssetProcessorConfig)
     # Disable platform in settings registry
     file_location = path.join(config_setreg_path, AssetProcessorConfig)
     with RegistrySettings(file_location) as settings:
         settings.set_key(f'{ASSET_PROCESSOR_SETTINGS_ROOT_KEY}/Platforms/{platform}', 'disabled')
 
 
-def disable_all_platforms(config_setreg_path):
```
Comment (Kadino) [inline Tools/LyTestTools/ly_test_tools/o3de/asset_processor_config_util.py:110]: ditto

File: Tools/LyTestTools/ly_test_tools/o3de/asset_processor_utils.py [hunk 8]
```diff
@@ -13,25 +13,12 @@
 import ly_test_tools.environment.process_utils as process_utils
 
 logger = logging.getLogger(__name__)
-processList = ["AssetProcessor_tmp","AssetProcessor","AssetProcessorBatch","AssetBuilder","rc","Lua Editor"]
-
-def start_asset_processor(bin_dir):
```
Comment (Kadino) [inline Tools/LyTestTools/ly_test_tools/o3de/asset_processor_utils.py:18]: ditto

**Review by Kadino — CHANGES_REQUESTED** [review]
A few points which may be resolved through discussion:

File: Tools/LyTestTools/ly_test_tools/o3de/asset_processor_config_util.py (hunk 6 above)
Comment (evanchia-ly-sdets) [reply Tools/LyTestTools/ly_test_tools/o3de/asset_processor_config_util.py:76]: This function has an undefined reference SUPPORTED_PLATFORMS so it doesn't even work.

File: Tools/LyTestTools/ly_test_tools/o3de/asset_processor_config_util.py (hunk 7 above)
Comment (evanchia-ly-sdets) [reply Tools/LyTestTools/ly_test_tools/o3de/asset_processor_config_util.py:110]: same undefined reference

File: Tools/LyTestTools/ly_test_tools/o3de/asset_processor_utils.py (hunk 8 above)
Comment (evanchia-ly-sdets) [reply Tools/LyTestTools/ly_test_tools/o3de/asset_processor_utils.py:18]: I searched the codebase for any calls of this and it was safe. Unfortunately the other functions in this file had a few usages which is why I didn't remove those.

File: Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py (hunk 1 above)
Comment (evanchia-ly-sdets) [reply Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py:114]: I don't mind pointing out when the exception occurs to make it more readable, but can add back the suggestion of a crash:
`Asset Processor has crashed or unexpectedly shut down during idle wait.`

File: Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py (hunk 3 above)
Comment (evanchia-ly-sdets) [reply Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py:212]: set back to debug

File: Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py (hunk 5 above)
Comment (evanchia-ly-sdets) [reply Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py:660]: Yeah the splitting of if/else string concats did not play nice. fixed to suggestion

**Review by cgalvan — APPROVED** [review]

**Review by Kadino — APPROVED** [review]

**Review by AMZN-scspaldi — APPROVED** [review]

**Pull request merged**